factorio-map-process.sh

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env bash
+set -euo pipefail
+## Defaults
+SERVER_BASE_PATH=/srv/factorio/maps
+MAP_TILES_PATH=images
+FACTORIO_MAP_SCRIPT="$(dirname "${BASH_SOURCE[0]}")/factoriomap.py"
+TARFILE=
+REDUCE=0
+
+function usage() {
+    cat <<EOF
+USAGE
+
+factorio-map-process.sh [OPTIONS] tarfile
+
+DESCRIPTION
+
+This script will process a given tarfile into a set of images used in a leaflet map.
+The default behavior will create a directory for the world maps if one does not exist and copy the example html page in as 'index.html'
+The tar file will be parsed for a map name and a date, separated by underscores (_). Files should be of the format {WorldName}_YYYY-MM-DD.tar.
+    
+OPTIONS
+
+    -t | --tiles <images>           Set the name of the directory to store the tile images in.  Default value is
+                                    'images'
+
+    -s | --server <server path>     Set the base path where the http server files will be stored.  Default value is
+                                    '/srv/factorio/maps'
+
+    -r                              Flag to size-reduce the images folder with the rdfind tool
+EOF
+}
+
+function testRequirements() {
+	if ! [ -x "$(command -v jq)" ]; then
+		echo "ERROR: Requirement not satisfied - jq -"
+		exit 1
+	fi
+
+        if [[ REDUCE -eq 1 && !(-x "$(command -v rdfind)") ]]; then
+		echo "ERROR: Requirement not satisfied - rdfind -"
+		exit 1
+	fi
+
+}
+
+function parseFileName() {
+	NAME_NOEXT="${FILENAME%.*}"
+	echo "Target File: ${FILENAME}"
+
+	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< $NAME_NOEXT))
```

Comment:
I tested it, this breaks if the filename has a space in it:
```bash
NAME_NOEXT="my world_2020-01-01"
infos=($(awk -F'[_.]' '{print $1; print $2}' <<< $NAME_NOEXT))
for i in $infos; do echo $i; done
my
world
2020-01-01
```
The `var=( some things )` form uses whitespace splitting to decide where to break into array values. If you use the default IFS var, it will separate the words around the space and the newlines printed by `awk`.
This should fix it:
```suggestion
	IFS='\n'
	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< "${NAME_NOEXT}"))
	unset IFS
```

See https://www.gnu.org/software/bash/manual/html_node/Word-Splitting.html